### PR TITLE
Add the option for a fallback AES key for decryption

### PR DESCRIPTION
**Pull request by mprahl**
This allows for a seamless decryption during a key rotation where there
is a temporary period that some values are encrypted with the old key
and some other values are encrypted with the new key.

Relates:
https://github.com/stolostron/backlog/issues/18709

### CODE REVIEW

File: pkg/templates/encryption.go [hunk 1]
```diff
@@ -50,30 +50,53 @@ func (t *TemplateResolver) protect(value string) (string, error) {
...
-		return "", ErrInvalidIV
+	var aesKeys [][]byte
+	if t.config.AESKeyFallback == nil {
+		aesKeys = [][]byte{t.config.AESKey}
+	} else {
+		aesKeys = [][]byte{t.config.AESKey, t.config.AESKeyFallback}
 	}
 
-	blockMode := cipher.NewCBCDecrypter(block, t.config.InitializationVector)
+	for _, aesKey := range aesKeys {
```
Comment (JustinKuli) [inline pkg/templates/encryption.go:74]: Should this have a `break` on a successful decryption? It seems like it always goes on to try the fallback key... but I would expect it to return the result at the first point where it successfully decrypts.

File: pkg/templates/templates_test.go [hunk 2]
```diff
@@ -315,6 +316,17 @@ func TestResolveTemplate(t *testing.T) {
 			"value: Raleigh",
 			nil,
 		},
+		{
+			"value: $ocm_encrypted:Eud/p3S7TvuP03S9fuNV+w==",
+			Config{
+				EncryptionConfig: EncryptionConfig{
+					AESKey: otherKey, AESKeyFallback: key, DecryptionEnabled: true, InitializationVector: iv,
```
Comment (JustinKuli) [inline pkg/templates/templates_test.go:323]: This test case shows it used the fallback key to decrypt, right? Should there be another test to ensure that it tries the "main" key, even if a fallback key was given?

**Review by JustinKuli — CHANGES_REQUESTED** [review]

File: pkg/templates/encryption.go [hunk 3]
```diff
@@ -50,30 +50,53 @@ func (t *TemplateResolver) protect(value string) (string, error) {
...
-	}
+	var decryptionErr error
+	var decryptedValue []byte
 
-	// This is already validated in the NewResolver method, but is checked again in case that method was bypassed
-	// to avoid a panic.
-	if len(t.config.InitializationVector) != IVSize {
-		return "", ErrInvalidIV
+	var aesKeys [][]byte
+	if t.config.AESKeyFallback == nil {
```
Comment (ChunxiAlexLuo) [inline pkg/templates/encryption.go:68]: So when both AESKey and AESKeyFallback are set, AESKeyFallback priority is higher?

File: pkg/templates/encryption.go (hunk 1 above)
Comment (mprahl) [reply pkg/templates/encryption.go:74]: Thanks for catching this! I meant to add it but forgot to submit it.

File: pkg/templates/templates_test.go (hunk 2 above)
Comment (mprahl) [reply pkg/templates/templates_test.go:334]: Good idea! I just added it.

File: pkg/templates/encryption.go (hunk 3 above)
Comment (mprahl) [reply pkg/templates/encryption.go:68]: It's not. If you look at how `aesKeys` is defined when `t.config.AESKeyFallback` is not `nil`, you'll see that `t.config.AESKey` is first in the `aesKeys` slice.

**Review by ChunxiAlexLuo — APPROVED** [review]
/lgtm

**Review by JustinKuli — APPROVED** [review]

**Pull request merged**